sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/filesystem/FileIndexer.java

```diff
@@ -63,24 +61,35 @@
   private final InputFileFilter[] filters;
   private final boolean isAggregator;
   private final ExclusionFilters exclusionFilters;
-  private final InputFileBuilderFactory inputFileBuilderFactory;
+  private final IndexedFileBuilder indexedFileBuilder;
+  private final MetadataGenerator metadataGenerator;
+  private final DefaultComponentTree componentTree;
+  private final DefaultInputModule module;
+  private final BatchIdGenerator batchIdGenerator;
+  private final InputComponentStore componentStore;
 
   private ProgressReport progressReport;
-  private ExecutorService executorService;
-  private List<Future<Void>> tasks;
 
-  public FileIndexer(ExclusionFilters exclusionFilters, InputFileBuilderFactory inputFileBuilderFactory, ProjectDefinition def, InputFileFilter[] filters) {
+  public FileIndexer(BatchIdGenerator batchIdGenerator, InputComponentStore componentStore, DefaultInputModule module, ExclusionFilters exclusionFilters,
+    DefaultComponentTree componentTree, IndexedFileBuilder indexedFileBuilder, MetadataGenerator inputFileBuilder, ProjectDefinition def, InputFileFilter[] filters) {
+    this.batchIdGenerator = batchIdGenerator;
+    this.componentStore = componentStore;
+    this.module = module;
+    this.componentTree = componentTree;
+    this.indexedFileBuilder = indexedFileBuilder;
+    this.metadataGenerator = inputFileBuilder;
     this.filters = filters;
     this.exclusionFilters = exclusionFilters;
-    this.inputFileBuilderFactory = inputFileBuilderFactory;
     this.isAggregator = !def.getSubProjects().isEmpty();
   }
 
-  public FileIndexer(ExclusionFilters exclusionFilters, InputFileBuilderFactory inputFileBuilderFactory, ProjectDefinition def) {
-    this(exclusionFilters, inputFileBuilderFactory, def, new InputFileFilter[0]);
+  public FileIndexer(BatchIdGenerator batchIdGenerator, InputComponentStore componentStore, DefaultInputModule module, ExclusionFilters exclusionFilters,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)